src/query/functions/aggregation/take.go

```diff
@@ -198,3 +257,38 @@ func takeFn(heap utils.FloatHeap, values []float64, buckets [][]int) []float64 {
 
 	return values
 }
+
+func takeInstantFn(heap *utils.FloatHeap, values []float64, buckets [][]int, metas []block.SeriesMeta) []valueAndMeta {
+	var result = make([]valueAndMeta, len(values))
+	for i := range result {
+		result[i] = valueAndMeta{
+			val:        values[i],
+			seriesMeta: metas[i],
+		}
+	}
+
+	for _, bucket := range buckets {
+		for _, idx := range bucket {
+			val := values[idx]
+			if !math.IsNaN(val) {
+				heap.Push(val, idx)
+			}
```

Comment:
Is fixing the NaN case within the scope of this work? i.e. these:

```
# Test NaN is sorted away from the top/bottom.
#eval_ordered instant at 50m topk(3, http_requests{job="api-server",group="production"})
#	http_requests{job="api-server", instance="1", group="production"}	200
#	http_requests{job="api-server", instance="0", group="production"}	100
#	http_requests{job="api-server", instance="2", group="production"}	NaN
```

Reply:
I think no, I would suggest to fix those later at some point.